daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/ImmArray.scala

```diff
@@ -367,8 +367,8 @@ object ImmArray {
   implicit val immArrayInstance: Traverse[ImmArray] = new Traverse[ImmArray] {
     override def traverseImpl[F[_], A, B](immArr: ImmArray[A])(f: A => F[B])(
         implicit F: Applicative[F]): F[ImmArray[B]] = {
-      F.map(immArr.foldLeft[F[BackStack[B]]](F.point(BackStack.empty)) {
-        case (ys, x) => F.apply2(ys, f(x))(_ :+ _)
+      F.map(immArr.foldLeft[F[BackStack[B]]](F.point(BackStack.empty)) { (ys, x) =>
```

Comment:
not a big fan of one-liners :) really hard to follow. I would have just assigned the value returned by `foldLeft` to a var and passed it to the `map`.

Reply:
Alas Scala inference disagrees with you :) I'll rewrite it to use the syntax extension, though.